File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/Adapter/CalendarAdapter.java
package com.byagowi.persiancalendar.Adapter;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.byagowi.persiancalendar.view.Fragment.CalendarMainFragment;
import com.byagowi.persiancalendar.view.Fragment.MonthNewFragment;

public class CalendarAdapter extends FragmentPagerAdapter {

    public CalendarAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        MonthNewFragment fragment = new MonthNewFragment();
        Bundle args = new Bundle();
        args.putInt("offset", position - CalendarMainFragment.MONTHS_LIMIT / 2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getCount() {
        return CalendarMainFragment.MONTHS_LIMIT;
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/adapter/CalendarAdapter.kt
package com.byagowi.persiancalendar.adapter

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.ViewPager
import com.byagowi.persiancalendar.Constants
import com.byagowi.persiancalendar.view.fragment.MonthFragment

class CalendarAdapter(fm: FragmentManager, isRTL: Boolean) : FragmentStatePagerAdapter(fm) {
  init {
    CalendarAdapter.isRTL = isRTL
  }

  override fun getItem(position: Int): Fragment {
    val fragment = MonthFragment()
    val bundle = Bundle()
    bundle.putInt(Constants.OFFSET_ARGUMENT, positionToOffset(position))
    fragment.arguments = bundle
    return fragment
  }

  override fun getCount(): Int = MONTHS_LIMIT

  companion object {
    private val MONTHS_LIMIT = 5000 // this should be an even number
    private var isRTL: Boolean = false

    fun gotoOffset(monthViewPager: ViewPager, offset: Int) {
      if (monthViewPager.currentItem != offsetToPosition(offset)) {
        monthViewPager.currentItem = offsetToPosition(offset)
      }
    }

    fun positionToOffset(position: Int): Int =
        if (isRTL) position - MONTHS_LIMIT / 2 else MONTHS_LIMIT / 2 - position

    private fun offsetToPosition(position: Int): Int =
        (if (isRTL) position else -position) + MONTHS_LIMIT / 2
  }
}
